runtime-light/stdlib/string/regex-functions.cpp

```diff
@@ -493,6 +530,83 @@ bool replace_regex(RegexInfo& regex_info, uint64_t limit) noexcept {
   return true;
 }
 
+std::optional<mixed> make_output_val(RegexInfo& regex_info, bool no_empty, bool offset_capture, size_t start_offset, size_t end_offset) noexcept {
+  const auto size{end_offset - start_offset};
+  if (no_empty && size == 0) {
+    return std::nullopt;
+  }
+
+  auto val{string{std::next(regex_info.subject.data(), start_offset), static_cast<string::size_type>(size)}};
+
+  mixed output_val;
+  if (offset_capture) {
+    output_val = array<mixed>::create(std::move(val), static_cast<int64_t>(start_offset));
+  } else {
+    output_val = std::move(val);
+  }
+
+  return output_val;
+}
+
+std::optional<array<mixed>> split_regex(RegexInfo& regex_info, int64_t limit_val, bool no_empty, bool delim_capture, bool offset_capture) noexcept {
+  auto offset{0uz};
+
+  if (limit_val == 0) {
+    limit_val = kphp::regex::PREG_NOLIMIT;
+  }
+
+  const auto& regex_state{RegexInstanceState::get()};
+  if (regex_info.regex_code == nullptr || !regex_state.match_context) [[unlikely]] {
```

Comment:
Let's follow the rule of early return. Here we can check `regex_code == nullptr` at the beginning of the function